ledger/participant-integration-api/src/main/scala/platform/store/backend/localstore/PartyRecordStorageBackendImpl.scala

```diff
@@ -105,14 +105,30 @@ object PartyRecordStorageBackendImpl extends PartyRecordStorageBackend {
     )
   }
 
-  override def updateIdentityProviderId(
-      internalId: Int,
+  override def fetchPartiesExist(
+      parties: Set[Ref.Party],
       identityProviderId: Option[IdentityProviderId.Id],
-  )(connection: Connection): Boolean =
-    IdentityProviderAwareBackend.updateIdentityProviderId("participant_party_records")(
-      internalId,
-      identityProviderId,
-    )(
-      connection
-    )
+  )(connection: Connection): Set[Ref.Party] = {
+    import com.daml.platform.store.backend.common.SimpleSqlAsVectorOf._
+    import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+    val filteredParties = if (parties.nonEmpty) {
+      cSQL"party in (${parties.map(_.toString)})"
+    } else {
+      cSQL"1 != 1"
```

Comment:
maybe someone knows a bit better approach here?

Reply:
that seems ok, we sometimes used "1 = 2"

Reply:
We have `com.daml.platform.store.backend.common.QueryStrategy#constBooleanWhere` and `com.daml.platform.store.backend.common.QueryStrategy#constBooleanSelect`